test.py

```diff
@@ -1,48 +1,50 @@
+def evaluate(dataloader: data.DataLoader, model: nn.Module):
+    device = next(model.parameters()).device
     model.eval()
     print("Evaluating...")
-    val_loss = []
-    val_acc = []
+    tp, tn, fp, fn = 0, 0, 0, 0
     with torch.no_grad():
-        for i, batch in enumerate(dataloader):
+        for i, batch in enumerate(tqdm(dataloader)):
             images, labels = batch
-            labels = torch.LongTensor(labels)
-            images, labels = images.to(device), labels.to(device)
-            outputs = model(images)
-            loss = loss_fn(outputs, labels)
-            acc = (torch.max(outputs.data, 1)[1] == labels).sum().item()
-            val_acc.append(acc.item())
-            val_loss.append(loss.item())
-    avg_loss = np.mean(val_loss)
-    avg_acc = np.mean(val_acc)
+            outputs = model(images.to(device))
+            outputs = outputs.cpu()
+            tn_batch, fp_batch, fn_batch, tp_batch = metrics.confusion_matrix(y_true=labels,
+                                                                              y_pred=torch.max(outputs.data, 1)[1],
+                                                                              labels=[0, 1]).ravel()
+            tp += tp_batch
+            tn += tn_batch
+            fp += fp_batch
+            fn += fn_batch
+    apcer = fp / (tn + fp) if fp != 0 else 0
+    bpcer = fn / (fn + tp) if fn != 0 else 0
+    acer = (apcer + bpcer) / 2
 
-    return avg_loss, avg_acc
+    return apcer, bpcer, acer
 
 
 if __name__ == '__main__':
     argparser = ArgumentParser()
     argparser.add_argument('--protocol', type=int, required=True)
-    argparser.add_argument('--data-dir', type=str, default=os.path.combine('data', 'CASIA_SURF'))
+    argparser.add_argument('--data-dir', type=str, default=os.path.join('data', 'CASIA_SURF'))
     argparser.add_argument('--checkpoint', type=str, required=True)
     argparser.add_argument('--num_classes', type=int, default=2)
     argparser.add_argument('--batch_size', type=int, default=1)
     args = argparser.parse_args()
-    dataset = CasiaSurfDataset(args.protocol, dir=args.data_dir,
-                               train=False, transform=transforms.Resize((320, 240)))
-    dataloader = data.DataLoader(dataset, batch_size=args.batch_size)
+    dataset = CasiaSurfDataset(args.protocol, dir=args.data_dir, transform=transforms.Resize((320, 240)))
+    dataloader = SplittedDataLoader(dataset, train_batch_size=1, val_batch_size=args.batch_size)
     model = models.mobilenet_v2(num_classes=args.num_classes)
-    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
-    model.load_state_dict(torch.load(args.checkpoint, map_location=device))
-    loss_fn = nn.CrossEntropyLoss()
-    avg_loss, avg_acc = evaluate(dataloader, model, loss_fn)
-    print(f'Average loss: {avg_loss}, Average Accuracy: {avg_acc}')
+    model.load_state_dict(torch.load(args.checkpoint, map_location='cpu'))

```

Comment:
It would be better to create variable `device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')` and use it here.